tests/unit/test_build.py

```diff
@@ -35,3 +36,23 @@ def test_get_snapcraft_build_plan(monkeypatch):
     assert fake_build_plan.mock_calls == [
         call(base="core22", build_base="core24", platforms={}, snap_type="base")
     ]
+
+
+@pytest.mark.parametrize(
+    ("app"),
+    [
+        *_build._APP_SPECIFIC_PLANNERS,
```

Comment:
This behaviour is undefined for Charmcraft. Strictly speaking Charms don't accept `base: bare`, but this isn't a craft-platforms concern.